pkg/feature/builder.go

```diff
@@ -27,24 +27,44 @@ type featureBuilder struct {
 	fsys     fs.FS
 }
 
-func CreateFeature(name string) *featureBuilder { //nolint:golint,revive //No need to export featureBuilder.
-	return &featureBuilder{
+func CreateFeature(name string) *featureName { //nolint:golint,revive //No need to export featureBuilder.
+	return &featureName{
 		name: name,
-		fsys: embeddedFiles,
 	}
 }
 
-func (fb *featureBuilder) For(spec *v1.DSCInitializationSpec) *featureBuilder {
+type featureName struct {
+	name string
+}
+
+func (fn *featureName) With(spec *v1.DSCInitializationSpec) *featureOrigin {
+	return &featureOrigin{
+		spec: spec,
+		name: fn.name,
+	}
+}
+
+type featureOrigin struct {
+	spec *v1.DSCInitializationSpec
+	name string
+}
+
+func (fo *featureOrigin) DefinedBy(origin featurev1.Origin) *featureBuilder {
```

Comment:
you mind renaming it from `DefinedBy(origin featurev1.Origin)` to `From(source featurev1.Source) `

personally speaking, DeinfedBy and origin are not very easy to understand.

Reply:
done here! 
1f7c936